mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -2220,48 +2219,67 @@ impl ElectrumClient {
 #[cfg_attr(test, mockable)]
 impl UtxoRpcClientOps for ElectrumClient {
     fn list_unspent(&self, address: &Address, _decimals: u8) -> UtxoRpcFut<Vec<UnspentInfo>> {
```

Comment:
An idea of another impl for pubkey balance support:
Maybe create a enum like AddressOrPubkey and pass a list of such enums into list_unspent. That is, normally we will pass both the ordinary p2pkh address and pubkey for my keypair. 
I think this would look as more explicit, so less side effects are possible

Reply:
Ummmmm, IMO, I think this would just take that abstraction a level higher.

I think documenting the method to tell that it will fetch the balance found in `p2pk` (if a `pubkey` was found) should be enough. The caller will be able to pick the UTXOs they are interested in since we provide `script_pubkey` in `UtxoInfo` now (they can select UTXOs with expected script pubkeys or filter out UTXOs with p2pk script pubkeys, etc...)

Reply:
In fact, original daemons have such an abstraction: the TxDestination type which is a selection of several variants (pubkey, p2pkh, p2sh,..).
I think yet maybe once we would like to combine segwit and non-segwit coins into one coin. If so, we will need to deal with several destinations for getting balance (but this is IMO)

Reply:
> I think yet maybe once we would like to combine segwit and non-segwit coins into one coin.

I would go for patching the `list_unspent` to fetch both balances in this case (instead of calling the method twice), and we can still separate segwit vs non-segwit `UtxoInfo`s at this point. (that's IMO as well :smiley:)

I think we could benefit from a third perspective on this one.

Reply:
> I think yet maybe once we would like to combine segwit and non-segwit coins into one coin

The problem with this is that we will need to support multiple addresses per coin to make the balance shown in our wallet same as the one in block explorers. This will require a substantial change in GUIs and UX too. So maybe we will call `list_unspent_group` if we did this.

I would keep the current implementation as is, since any refactor in this PR might change in the future.